diff -uwbr samba-3.0.25-org//source/configure samba-3.0.25/source/configure
--- samba-3.0.25-org//source/configure	2009-07-17 08:43:20.000000000 +0200
+++ samba-3.0.25/source/configure	2009-07-17 09:31:56.000000000 +0200
@@ -43429,7 +43429,7 @@
 
 
 
-for ac_header in linux/inotify.h asm/unistd.h
+for ac_header in linux/inotify.h asm/unistd.h sys/inotify.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
diff -uwbr samba-3.0.25-org//source/configure.in samba-3.0.25/source/configure.in
--- samba-3.0.25-org//source/configure.in	2007-05-11 00:09:38.000000000 +0200
+++ samba-3.0.25/source/configure.in	2009-07-17 09:31:56.000000000 +0200
@@ -2513,7 +2513,7 @@
 fi
 
 AC_CACHE_CHECK([for inotify support],samba_cv_HAVE_INOTIFY,[
-AC_CHECK_HEADERS(linux/inotify.h asm/unistd.h)
+AC_CHECK_HEADERS(linux/inotify.h asm/unistd.h sys/inotify.h)
 AC_CHECK_FUNC(inotify_init)
 AC_HAVE_DECL(__NR_inotify_init, [#include <asm/unistd.h>])
 ],
Only in samba-3.0.25-org//source: dynconfig.o
Only in samba-3.0.25-org//source/groupdb: mapping.o
Only in samba-3.0.25-org//source/groupdb: mapping_tdb.o
Only in samba-3.0.25-org//source/include: build_env.h
diff -uwbr samba-3.0.25-org//source/smbd/notify_inotify.c samba-3.0.25/source/smbd/notify_inotify.c
--- samba-3.0.25-org//source/smbd/notify_inotify.c	2007-05-11 00:09:34.000000000 +0200
+++ samba-3.0.25/source/smbd/notify_inotify.c	2009-07-17 10:00:25.000000000 +0200
@@ -25,15 +25,21 @@
 #include "includes.h"
 
 #ifdef HAVE_INOTIFY
+#define HAVE_SYS_INOTIFY_H 1
+#if HAVE_SYS_INOTIFY_H
+#include <sys/inotify.h>
+#else
 
 #ifdef HAVE_ASM_TYPES_H
 #include <asm/types.h>
 #endif
 
+#ifndef HAVE_INOTIFY_INIT
+
 #include <linux/inotify.h>
 #include <asm/unistd.h>
 
-#ifndef HAVE_INOTIFY_INIT
+
 /*
   glibc doesn't define these functions yet (as of March 2006)
 */
@@ -51,8 +57,12 @@
 {
 	return syscall(__NR_inotify_rm_watch, fd, wd);
 }
-#endif
+#else
+
+#include <sys/inotify.h>
 
+#endif
+#endif
 
 /* older glibc headers don't have these defines either */
 #ifndef IN_ONLYDIR
